### PR TITLE
feat: Add validation for import aliases

### PR DESCRIPTION
**Pull request by justasam**
Closes #<issue>

## ✅ Checklist

- [X] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [X] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [X] I performed a functional test on my final commit

---

## Changelog

- Add validation for import aliases

---

## Screenshots

image

💯

### CODE REVIEW

**Review by juliusmarminge — CHANGES_REQUESTED** [review]
Please use the `validate` prop instead of calling the prompter multipel times, see `promptAppName`

**Review by c-ehrlich — APPROVED** [review]
Looks good now. Thanks for the contribution!

File: cli/src/utils/validateImportAlias.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const validateImportAlias = (input: string) => {
+  if (input.startsWith(".") || input.startsWith("/")) {
```
[the remark this answers is not in the document]
Comment (justasam) [reply cli/src/utils/validateImportAlias.ts:2]: I am not sure I understand -- what do you mean by it follows `/*`? 
AFAIK, the only thing not allowed is import alias starting with `.` or `/` (see [here](https://www.npmjs.com/package/tsconfig-paths)). And for that case regex is overkill - it just makes code harder to read.

[the remark this answers is not in the document]
Comment (juliusmarminge) [reply cli/src/utils/validateImportAlias.ts:2]: Oh ok. I thought they had to have the ending `/*` so for exmple `abc/*` would be valid but just `abc` wouldn't.

> And for that case regex is overkill - it just makes code harder to read.

Totally agree if that's the case!

**Review by juliusmarminge — APPROVED** [review]

File: cli/src/utils/validateImportAlias.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (justasam) [reply cli/src/utils/validateImportAlias.ts:2]: Ah, I understand what you've meant. That's handled by `cli/src/helpers/setImportAlias.ts` though, so `abc` will become `abc/` in imports and `abc/*` in tsconfig: 
```TSX
 const normalizedImportAlias = importAlias
    .replace(/\*/g, "") // remove any wildcards (~/* -> ~/)
    .replace(/[^\/]$/, "$&/"); // ensure trailing slash (@ -> ~/)
```

**Pull request merged**